### PR TITLE
User/michaelshikh/11 11 2022 release

### PR DESCRIPTION
**Pull request by michaelshikh07**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs [hunk 1]
```diff
@@ -547,6 +549,19 @@ public partial class AcceptedAudience
         public static bool operator !=(Azure.ResourceManager.Kusto.Models.BlobStorageEventType left, Azure.ResourceManager.Kusto.Models.BlobStorageEventType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class CosmosDbDataConnection : Azure.ResourceManager.Kusto.KustoDataConnectionData
+    {
+        public CosmosDbDataConnection() { }
+        public string CosmosDbAccountResourceId { get { throw null; } set { } }
```
Comment (Yao725) [inline sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs:555]: Could you please double confirm the type of this property? Is it an arm id, which will be formatted like `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}`? We need specific types in .NET SDK and [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs) will be used here if the property is an arm id.

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs [hunk 2]
```diff
@@ -547,6 +549,19 @@ public partial class AcceptedAudience
...
+        public string CosmosDbContainer { get { throw null; } set { } }
+        public string CosmosDbDatabase { get { throw null; } set { } }
+        public string ManagedIdentityObjectId { get { throw null; } }
+        public string ManagedIdentityResourceId { get { throw null; } set { } }
```
Comment (Yao725) [inline sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs:559]: Same question as above.

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1 [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿Import-Module Az.Kusto
```
Comment (Yao725) [inline sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1:1]: I don't think we want this script here. There are two reasons, one is that the test framework will clean up the created resource by default, second is that it's better to keep the folder structure neat and clear. You can save this locally, but there's no need to merge it into the repo.

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/RunTestsInRecordMode.ps1 [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿Write-Host "Going to run test in record mode, please make sure required resources exists, if not, please run 'CreateTestResurces.ps1'" -ForegroundColor Green
```
Comment (Yao725) [inline sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/RunTestsInRecordMode.ps1:1]: same as above

File: sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md [hunk 5]
```diff
@@ -1,15 +1,17 @@
 # Release History
 
-## 1.2.0-beta.1 (Unreleased)
+## 1.2.0 (2023-01-03)
```
Comment (ArthurMa1978) [inline sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md:3]: Please give 2 weeks ahead at least to release a new package

Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md:3]: @ArthurMa1978 
2 weeks is (2023-01-17)? 
Can you please explain why the 2 weeks ahead is needed?
How soon can I get a dev feed nugget?

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CreateTestResources.ps1 [hunk 6]
```diff
@@ -0,0 +1,38 @@
+﻿Import-Module Az.Kusto
```
[the remark this answers is not in the document]
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CreateTestResources.ps1:1]: @Yao725 
It is replacing setup.ps1 (removed in this pr)
Is it OK if I put all my scripts in Test directory (where setup.ps1 used to be) ?

[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CreateTestResources.ps1:1]: Ok, I think it's fine to remove the scripts to the test directory, please make sure that you also update this [file](https://github.com/Azure/azure-sdk-for-net/blob/4f871a3f3d7fe6b7654f28f7f115420754e6afa2/sdk/kusto/Azure.ResourceManager.Kusto/tests/Prerequisites/prerequisites.readme.md) so we can better maintain these test scripts in the future.

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1 (hunk 3 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1:1]: I have to push it to the repository, 
I want those scripts to be shared across all my team members,
I am not going to be the only one on my team in charge of releasing sdk for kusto.

let me know in which directory can I put my scripts (like setup.ps1) 

by the way we had leftovers from previous c# sdk release not weren't cleared 
I had to clear them manually, this is why I have crated this cleanup script

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs [hunk 7]
```diff
@@ -547,6 +549,19 @@ public partial class AcceptedAudience
         public static bool operator !=(Azure.ResourceManager.Kusto.Models.BlobStorageEventType left, Azure.ResourceManager.Kusto.Models.BlobStorageEventType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class CosmosDbDataConnection : Azure.ResourceManager.Kusto.KustoDataConnectionData
+    {
+        public CosmosDbDataConnection() { }
+        public string CosmosDbAccountResourceId { get { throw null; } set { } }
+        public string CosmosDbContainer { get { throw null; } set { } }
+        public string CosmosDbDatabase { get { throw null; } set { } }
+        public string ManagedIdentityObjectId { get { throw null; } }
```
[the remark this answers is not in the document]
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs:558]: CosmosDbAccountResourceId  is resource identifier 
CosmosDbContainer is a name (string)
CosmosDbDatabase is name (string)
ManagedIdentityObjectId is GUID

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs (hunk 1 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs:555]: CosmosDbAccountResourceId  is /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName} format.

how do you want me to declare it? 
is it auto generated from swagger?
How do prevent override upon next generation ?

File: sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md (hunk 5 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md]: updated to 2023-01-17

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs:558]: I have added mapping to autorest.md to specify the exact type

File: sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs (hunk 2 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/api/Azure.ResourceManager.Kusto.netstandard2.0.cs]: I have added mapping to autorest.md

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CreateTestResources.ps1 (hunk 6 above)
[the remark this answers is not in the document]
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CreateTestResources.ps1]: moved all scripts to test path and updated readme file as agreed

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/RunTestsInRecordMode.ps1 (hunk 4 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/RunTestsInRecordMode.ps1]: moved all scripts to test path and updated readme file as agreed

File: sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1 (hunk 3 above)
Comment (michaelshikh07) [reply sdk/kusto/Azure.ResourceManager.Kusto/src/Scripts/CleanupTestResources.ps1]: moved all scripts to test path and updated readme file as agreed

**Review by Yao725 — APPROVED** [review]

**Pull request merged**